sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,217 @@
+# Author: Henry Lin <hlin117@gmail.com>
+
+from __future__ import division
+
+import numpy as np
+from ..externals.six.moves import xrange
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_array, check_is_fitted, column_or_1d
+
+
+class KBinsDiscretizer(BaseEstimator, TransformerMixin):
+    """Bins continuous data into k equal width intervals.
+
+    Parameters
+    ----------
+    n_bins : int (default=2)
```

Comment:
do you think we should support one or more automatic determination methods of `numpy.histogram`?
do you think we should support different numbers of bins for different features (which is entailed by considering some of the `numpy.histogram` methods Auto, DF, Doane and Scott)?